confidence.py

```diff
@@ -237,7 +237,62 @@ def loads(*strings):
     return Configuration(*(yaml.load(string) for string in strings))
 
 
-def read_envvars(name):
+def read_xdg_config_dirs(name, extension):
+    """
+    Read from files found in XDG-specified system-wide configuration paths,
+    defaulting to ``/etc/xdg``. Depends on ``XDG_CONFIG_DIRS`` environment
+    variable.
+
+    :param name: application or configuration set name
+    :param extension: file extension to look for
+    :return: a `.Configuration` instance with values read from XDG-specified
+        directories
+    """
+    # find optional value of ${XDG_CONFIG_DIRS}
+    config_dirs = environ.get('XDG_CONFIG_DIRS')
+    if config_dirs:
+        # PATH-like env vars operate in decreasing precedence, reverse this path set to mimic the end result
+        config_dirs = reversed(config_dirs.split(path.pathsep))
+    else:
+        # XDG spec: "If $XDG_CONFIG_DIRS is either not set or empty, a value equal to /etc/xdg should be used."
+        config_dirs = ['/etc/xdg']
+
+    # collect existing files in the config dirs
+    hits = []
+    for config_dir in config_dirs:
+        candidate = path.join(config_dir, '{name}.{extension}'.format(name=name, extension=extension))
+        if path.exists(candidate):
+            hits.append(candidate)
+
+    return loadf(*hits)
+
+
+def read_xdg_config_home(name, extension):
+    """
+    Read from file found in XDG-specified configuration home directory,
+    expanding to ``${HOME}/.config/name.extension`` by default. Depends on
+    ``XDG_CONFIG_HOME`` or ``HOME`` environment variables.
+
+    :param name: application or configuration set name
+    :param extension: file extension to look for
+    :return: a `.Configuration` instance, possibly `.NotConfigured`
+    """
+    # find optional value of ${XDG_CONFIG_HOME}
+    config_home = environ.get('XDG_CONFIG_HOME')
+    if not config_home:
+        # XDG spec: "If $XDG_CONFIG_HOME is either not set or empty, a default equal to $HOME/.config should be used."
+        # see https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html
+        config_home = path.expanduser('~/.config')
+
+    # expand to full path to configuration file in XDG config path
+    config_path = path.join(config_home, '{name}.{extension}'.format(name=name, extension=extension))
+    if path.exists(config_path):
```

Comment:
It's a matter of preference, but I'd test for _non_-existence here, so the function ends with the load of the config, instead of returning NonConfigured.

Reply:
Good point, makes the function align with the other `read_*` friends. Also aligned `read_envvars` to do this and return a `Configuration` instance.